File: examples/flutter/lib/src/models/team_list.dart
part of appwrite.models;

/// Teams List
class TeamList implements Model {
    /// Total sum of items in the list.
    final int sum;
    /// List of teams.
    final List<Team> teams;

    TeamList({
        required this.sum,
        required this.teams,
    });

    factory TeamList.fromMap(Map<String, dynamic> map) {
        return TeamList(
            sum: map['sum'],
            teams: List<Team>.from(map['teams'].map((p) => Team.fromMap(p))),
        );
    }

    @override
    Map<String, dynamic> toMap() {
        return {
            "sum": sum,
            "teams": teams.map((p) => p.toMap()),
        };
    }
}

File: examples/swift-server/Sources/AppwriteModels/TeamList.swift

/// Teams List
public class TeamList {

    /// Total number of items available on the server.
    public let sum: Int

    /// List of teams.
    public let teams: [Team]

    init(
        sum: Int,
        teams: [Team]
    ) {
        self.sum = sum
        self.teams = teams
    }

    public static func from(map: [String: Any]) -> TeamList {
        return TeamList(
            sum: map["sum"] as! Int,
            teams: (map["teams"] as! [[String: Any]]).map { Team.from(map: $0) }
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "sum": sum as Any,
            "teams": teams.map { $0.toMap() } as Any
        ]
    }
                                                                                                                                                                    
}